### DISCUSSION TITLE
All of the sudden - Unable to load Sprites, Uncaught (in promise) ImageResource.ts:182

### DISCUSSION THREAD

**Discussion by johnkallen** [pixijs/pixijs#8324] [Q&A]
### Expected Behavior
Loading Sprites individually or from sprite sheet

### Current Behavior
Made no changes to my code, code that I haven't touched for days fails to load sprites
Get this ERROR:
Uncaught (in promise) Event {isTrusted: true, type: 'error', target: null, currentTarget: null, eventPhase: 0, …}
source.onerror @ ImageResource.ts:182
error (async)
(anonymous) @ ImageResource.ts:179
ImageResource.load @ ImageResource.ts:144
ImageResource @ ImageResource.ts:123
autoDetectResource @ autoDetectResource.ts:119
BaseTexture @ BaseTexture.ts:251
Texture.from @ Texture.ts:389
Sprite.from @ Sprite.ts:516
./src/GameTutorial.js @ GameTutorial.js:104
options.factory @ react refresh:6
__webpack_require__ @ bootstrap:24
fn @ hot module replacement:62
./src/index.js @ GameTutorial.js:200
options.factory @ react refresh:6
__webpack_require__ @ bootstrap:24
(anonymous) @ startup:7
(anonymous) @ startup:7

### Possible Solution
Tried loading many Sprites MANY different PIXI ways, all FAILED.

### Steps to Reproduce
(Multiple ways)
const playerSprite = new PIXI.Sprite.from('../images/player.png'); <-- this line causes the error
app.stage.addChild(playerSprite);

### Environment
"pixi.js": "^6.3.0",
"@pixi/graphics-extras": "^6.3.0",
Chrome Version 100.0.4896.127 (Official Build) (x86_64)
OS Mac: 12.3.1 (21E258)

- **`pixi.js` version**: _e.g. 4.7.1_
- **Browser & Version**: _e.g. Chrome 67_
- **OS & Version**: _e.g. Ubuntu 18.04_
- **Running Example**: _url_

**Comment by bigtimebuddy** [comment]
Please try

```js
const playerSprite = PIXI.Sprite.from('../images/player.png');
```
 Not: 
```js
const playerSprite = new PIXI.Sprite.from('../images/player.png');
```

**Comment by johnkallen** [comment]
LOL, Did something change with your code base last night? 

I was loading Sprites yesterday no problem, now NONE of them work. All these code examples give the same error above:

const playerSprite = PIXI.Sprite.from('./images/anyImage.png');

const loader = PIXI.Loader.shared;
loader
  .add("invoice", "../images/icon.png")
  .load();

const loader = PIXI.Loader.shared;
loader
  .add('tileset', '../images/woman.json')
  .load(setup);

function setup(loader, resources) {
  const womanWaveTexture = PIXI.Texture.from('__office_worker_wave_000.png');
  const womanWaveSprite = new PIXI.Sprite(womanWaveTexture);
  womanWaveSprite.position.set(100,100);
  womanWaveSprite.scale.set(0.4,0.4);
  app.stage.addChild(womanWaveSprite);
  const womanWaveTexture2 = new PIXI.Texture.from('__office_worker_wave_008.png');
  const womanWaveSprite2 = new PIXI.Sprite(womanWaveTexture2);
  womanWaveSprite2.position.set(200,100);
  womanWaveSprite2.scale.set(0.4,0.4);
  app.stage.addChild(womanWaveSprite2);
}

Something changed, these all worked yesterday.

**Comment by bigtimebuddy** [comment]
Can you please provide a running example (codepen.com, jsfiddle.net, pixiplayground.com)? [This example](https://pixijs.io/examples/#/demos-basic/container.js) works fine and uses `PIXI.Texture.from` running on the latest code.

**Comment by miltoncandelero** [comment]
> `const womanWaveTexture2 = new PIXI.Texture.from('__office_worker_wave_008.png');`

Do not mix `new` with `.from(...)` 😰

**Comment by johnkallen** [comment]
### Explain this:

This works **without error**:

import image_url from './images/myImage.png';
const texture = PIXI.Texture.from(image_url);

This causes the **Error stated above**:

const texture = PIXI.Texture.from('./images/myImage.png');

However, this **used to work**, I can't figure out what changed to all of the sudden make it NOT work..

**Comment by bigtimebuddy** [comment]
What you’ve described is webpack behavior. You should try to understand more how webpack handles images. Converting this to discussion as it seems more support related.

**Reply by johnkallen** [reply]
I'm not using webpack, I'm running this in React using Node on my local machine.

**Reply by johnkallen** [reply]
here is my entire package.json:

{
  "name": "pay-invaders",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@testing-library/jest-dom": "^5.16.4",
    "@testing-library/react": "^13.1.1",
    "@testing-library/user-event": "^13.5.0",
    "ajv": "^8.11.0",
    "pixi.js": "^6.3.2",
    "react": "^18.1.0",
    "react-dom": "^18.1.0",
    "react-pixi-fiber": "^0.14.3",
    "react-scripts": "5.0.1",
    "typescript": "^4.6.4",
    "web-vitals": "^2.1.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}

**Reply by johnkallen** [reply]
Only loading textures throws an error, I can create the app, change the background color, add containers, add text and format text via PIXI.TextStyle. ONLY texture loading does not work, I can't load sprite sheets or individual images. All this was working a few days ago and then all of the sudden, code that had been working has stopped working. 

I had an animated sprite that was working, made NO CHANGES to that code, but that sprite sheet now won't load.

I've started completely over with a new React project, but as soon as I get to a texture load like this:

const texture = PIXI.Texture.from('./images/myImage.png');

it throws that error.

**Reply by domis86** [reply]
> code that had been working has stopped working.

Which Pixi version did you used before (when it was working) and which version now (that is not working)?

**Reply by bigtimebuddy** [reply]
React scripts uses webpack. You need to [import images](https://create-react-app.dev/docs/adding-images-fonts-and-files/) not inline the URL (unless you're manually copying static assets to your dist folder).

**Reply by EnslavedTuna** [reply]
I have the same issue as described above, an old project that worked previously does not work anymore. The best example is actually the tiny demo from https://pixijs.io/guides/basics/getting-started.html under "Putting it all together" header. As the code as shown does currently not work anymore (uncaught promise error because you try and load sample.png directly)

**Comment by johnkallen** [comment]
I could never get it to work, I eventually dropped this pixijs library
altogether and switched to directly updating the HTML 5 Canvas, which
actually gave me more control and a faster frame rate. :)

Check out this youtube video:
https://youtu.be/gCa0z4B-CRo

**Reply by domis86** [reply]
> actually gave me more control and a faster frame rate. :)

How do you know that you have more control or frame rate is faster if you didnt compare it with Pixi.js? (as you said: "I could never get it to work")
